### ISSUE TITLE
attribute `[[noreturn]]` Diagnoses 'should not return' on a void function

### ISSUE THREAD

**Issue by erichkeane** [llvm/llvm-project#81016]
See https://godbolt.org/z/h5nsj8qEK

```
[[noreturn]]
void foo() {
}
```
```
<source>:3:1: warning: function declared 'noreturn' should not return [-Winvalid-noreturn]
    3 | }
      | ^
```
We should probably suppress this diagnostic when the function is void .

**Comment by Endilll** [comment]
I don't think it's a false-positive, as noreturn are not supposed to return, implicitly or explicitly (http://eel.is/c++draft/dcl.attr.noreturn#2):
> If a function f is called where f was previously declared with the noreturn attribute and f eventually returns, the behavior is undefined[.](http://eel.is/c++draft/dcl.attr.noreturn#2.sentence-1)

**Comment by erichkeane** [comment]
Ah, you're right.  It is the 'reaching end of scope' which causes this to be a return.  Thanks, closing.
